packages/lesswrong/components/jargon/GlossarySidebar.tsx

```diff
@@ -246,15 +287,40 @@ const GlossarySidebar = ({post, postGlossariesPinned, togglePin, showAllTerms, s
     </div>
   </LWTooltip>;
 
+
+  const titleRow = !onlyUnapprovedTerms ? (
+    <LWTooltip
+      title={tooltip}
+      inlineBlock={false}
+      placement='top-end'
+      popperClassName={classes.titleRowTooltipPopper}
+    >
+      <div className={classes.titleRow} onClick={() => togglePin('clickGlossaryContainer')}>
+        <h3 className={classes.title}>
+          Glossary
+        </h3> 
+        <ForumIcon icon="Dictionary" className={classNames(classes.pinIcon, displayAsPinned && classes.pinnedPinIcon)} />
+      </div>
+    </LWTooltip>
+  ) : (
+    <LWTooltip title={<div>Pin to {showAllTerms ? 'hide' : 'show'} a hacky AI generated glossary<br/> that the author doesn't endorse<div><em>(Opt/Alt + Shift + G)</em></div></div>} inlineBlock={false} placement='top-end' popperClassName={classes.titleRowTooltipPopper}>
+      <div className={classes.titleRow} onClick={(e) => setShowAllTerms(e, !showAllTerms, 'unapprovedGlossaryClick')}  {...unapprovedHoverHandlers}>
+        <ForumIcon icon="Dictionary" className={classNames(classes.pinIcon, classes.unapprovedPinIcon, showAllTerms && classes.pinnedPinIcon)} /> 
```

Comment:
All the instances of `showAllTerms` used for conditional styling here should also be `displayAsPinned`